src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/index.tsx

```diff
@@ -87,19 +86,10 @@ export const CreateNSTreeModal = ({
     if (shouldReset) setShouldReset(false);
   }, [shouldReset]);
 
-  useEffect(() => {
-    if (treeType !== undefined && Object.values(TreeTypes).includes(treeType)) {
```

Comment:
I just want to double-check: we're removing the check that the provided `treeType` is actually one of the existing TreeTypes we have defined, right? I don't see this moved over to a new runtime check, although the check is still mostly present because of the downstream Props type check with `treeType: TreeType | undefined`.

I am totally fine with this -- I think it's good and reasonable! -- but wanted to make sure I am reading things correctly.